src/main/webapp/js/common.js

```diff
@@ -484,20 +484,22 @@ function getPointValue(s, ditchZero) {
 /**
  * Checks if element is within browser's viewport.
  * @return true if it is within the viewport, false otherwise
+ * @see http://stackoverflow.com/q/123999
  */
 function isWithinView(element) {
-    var viewHeight = window.innerHeight;
-    var viewTop = window.scrollY;
-    var viewBottom = viewTop + viewHeight;
-    
-    var elementHeight = element.offsetHeight;
-    var elementTop = element.offsetTop;
-    var elementBottom = elementTop + elementHeight;
+    var baseElement = $(element)[0]; // unwrap jquery element
+    var rect = baseElement.getBoundingClientRect();
+
+    var windowHeight = window.innerHeight || document.documentElement.clientHeight;
+    var windowWidth = window.innerWidth || document.documentElement.clientWidth;
     
-    return viewHeight >= elementHeight
-           ? viewTop <= elementTop && viewBottom >= elementBottom          // all within view
-           : viewTop <= elementTop && viewBottom >= elementTop             // top within view
-             || viewTop <= elementBottom && viewBottom >= elementBottom;   // btm within view
+    var tolerance = 0.25;
```

Comment:
Sure, although it's arbitrary, I've taken a looked at the ranges of values that make the tests fail on firefox.
```
rect.left + rect.width
> 1620.000015258789
windowWidth
> 1620
```

So the difference is really minuscule, but the tests still fail. Having it as a percentage isn't very good as the range could be really big for 4K screens.